### PR TITLE
Updated Omnibus install steps

### PR DESCRIPTION
**Pull request by cwarnermm**
Updated Omnibus installation documentation for visitors coming directly from mattermost.com/download to the product documentation.

Added 3 temporary CTA placeholders for whose look/feel/code will be replaced by Marketing:
- adding the PPA repos
- installing MM Omnibus
- updating MM Omnibus

### CODE REVIEW

**Review by nab-77 — APPROVED** [review]
Looks great.

File: source/install/installing-mattermost-omnibus.rst [hunk 1]
```diff
@@ -4,106 +4,106 @@ Install Mattermost Omnibus
...
 
-After modifying the ``mmomni.yml`` configuration file, you need to run ``mmomni reconfigure`` for Omnibus to apply the changes and restart Mattermost.
+In a terminal window, run the following command to configure the repositories needed for a PostgreSQL database, configure an NGINX web server to act as a proxy, configure certbot to issue and renew the SSL certificate, and configure the Mattermost Omnibus repository so that you can run the install command.
 
-Use a custom NGINX template
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+``curl -o- https://deb.packages.mattermost.com/repo-setup.sh | sudo bash``  |copy-to-clipboard|
 
-Mattermost Omnibus generates an ``nginx`` configuration depending on how the different properties of the ``mmomni.yml`` file are set. However, you may need to customize the configuration further to support other use cases, such as using custom SSL certificates. For those cases, Omnibus supports using a custom ``nginx`` template to generate its configuration. 
+Step 2: Install Omnibus
```
Comment (cwarnermm) [inline source/install/installing-mattermost-omnibus.rst:30]:
```suggestion
Step 2: Install Mattermost Omnibus
```

File: source/install/installing-mattermost-omnibus.rst [hunk 2]
```diff
@@ -4,106 +4,106 @@ Install Mattermost Omnibus
...
-After modifying the ``mmomni.yml`` configuration file, you need to run ``mmomni reconfigure`` for Omnibus to apply the changes and restart Mattermost.
+In a terminal window, run the following command to configure the repositories needed for a PostgreSQL database, configure an NGINX web server to act as a proxy, configure certbot to issue and renew the SSL certificate, and configure the Mattermost Omnibus repository so that you can run the install command.
 
-Use a custom NGINX template
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+``curl -o- https://deb.packages.mattermost.com/repo-setup.sh | sudo bash``  |copy-to-clipboard|
 
-Mattermost Omnibus generates an ``nginx`` configuration depending on how the different properties of the ``mmomni.yml`` file are set. However, you may need to customize the configuration further to support other use cases, such as using custom SSL certificates. For those cases, Omnibus supports using a custom ``nginx`` template to generate its configuration. 
+Step 2: Install Mattermost Omnibus
+-----------------------
```
Comment (cwarnermm) [inline source/install/installing-mattermost-omnibus.rst:31]:
```suggestion
---------------------------------------
```

File: source/install/installing-mattermost-omnibus.rst [hunk 3]
```diff
@@ -4,106 +4,106 @@ Install Mattermost Omnibus
...
 
-If you wish to remove Mattermost and Mattermost Omnibus completely for any reason, you can run this command:
+3. Open a browser and navigate to your Mattermost domain either by domain name (e.g. mymattermostserver.com), or by the server’s IP address if you’re not using a domain name.
 
-``sudo apt remove --purge mattermost mattermost-omnibus``
+4. Create your first Mattermost user, invite more users, and explore the Mattermost platform.
 
-Backup and Restore using the Mattermost Omnibus CLI
----------------------------------------------------
+Step 3: Configure Omnibus
```
Comment (cwarnermm) [inline source/install/installing-mattermost-omnibus.rst:47]:
```suggestion
Step 3: Configure Mattermost Omnibus
```

File: source/install/installing-mattermost-omnibus.rst [hunk 4]
```diff
@@ -4,106 +4,106 @@ Install Mattermost Omnibus
...
-If you wish to remove Mattermost and Mattermost Omnibus completely for any reason, you can run this command:
+3. Open a browser and navigate to your Mattermost domain either by domain name (e.g. mymattermostserver.com), or by the server’s IP address if you’re not using a domain name.
 
-``sudo apt remove --purge mattermost mattermost-omnibus``
+4. Create your first Mattermost user, invite more users, and explore the Mattermost platform.
 
-Backup and Restore using the Mattermost Omnibus CLI
----------------------------------------------------
+Step 3: Configure Mattermost Omnibus
+-------------------------
```
Comment (cwarnermm) [inline source/install/installing-mattermost-omnibus.rst:48]:
```suggestion
-----------------------------------
```

**Comment by nab-77** [comment]
Wondering if we should remove the "step 1, step 2" etc. The current step 3 of configure is completely optional as the base config is done via the install.

File: source/install/installing-mattermost-omnibus.rst [hunk 5]
```diff
@@ -4,106 +4,106 @@ Install Mattermost Omnibus
...
 
-If you wish to remove Mattermost and Mattermost Omnibus completely for any reason, you can run this command:
+3. Open a browser and navigate to your Mattermost domain either by domain name (e.g. mymattermostserver.com), or by the server’s IP address if you’re not using a domain name.
 
-``sudo apt remove --purge mattermost mattermost-omnibus``
+4. Create your first Mattermost user, invite more users, and explore the Mattermost platform.
 
-Backup and Restore using the Mattermost Omnibus CLI
----------------------------------------------------
+Step 3: Configure Mattermost Omnibus
```
Comment (nab-77) [inline source/install/installing-mattermost-omnibus.rst:47]:
```suggestion
Configure Mattermost Omnibus
```

**Review by nab-77 — CHANGES_REQUESTED** [review]
Only 2 steps as configure options are optional

**Review by nab-77 — APPROVED** [review]

**Comment by emdecr** [comment]
Hey folks~

Pushed up my first pass for the addition of the copy buttons for the three instances @cwarnermm outlined.

The copy function is not working on the preview sites because[ the Clipboard API is only available in secure contexts (HTTPS) for most browsers](https://developer.mozilla.org/en-US/docs/Web/API/Clipboard).

I'd suggest getting things running locally to see what that all looks like.

Re: tracking -
The event we're pushing to the data layer looks like this:
``` js
dataLayer.push({
    event: 'copy.installation',
    installMethod: copyMethod,
    installCommand: copyCommand
});
```
`copyMethod` and `copyCommand` variables pull from data attributes on the block element. `copyMethod` is `Omnibus` for this page and I used the title under which each command sits for the command label, ex:

``` js
event: "copy.installation"
installCommand: "Add the Mattermost PPA repositories"
installMethod: "Omnibus"
```
cc: @aaron-l-thompson @nat-gunner Lemme know if all the tracking details are clear or if you need tweaks to get everything set up on your ends.

Definitely let me know if y'all have any questions, qualms, or feedback. Thanks!

**Comment by cwarnermm** [comment]
@emdecr - Love the new code block format on this page. I'd love to use this approach across all of the install docs pages (aligned to the /download webpage). 

The new CTA buttons are bigger than the Contact Sales CTA in the header. Thoughts on reducing their size slightly? Also, users can copy via the code block as well as the CTA. Is the event action being tracked regardless of how the command is copied to the clipboard?

**Comment by emdecr** [comment]
@cwarnermm I'll rope in @aaron-l-thompson re: button sizing. It's definitely a substantial size difference and draws your eye in – depending on what we want the main action to be on this page, it might make a difference?

And yes, I can definitely help our re: flooding this across the other install pages if you need a hand. 

Re: tracking - yep, the tracking is present for both when the user clicks on the code block or the button. We have a flag in the event that is sent for which method they use; note the `copyAction` item in the event:
```js
event: "copy.installation"
installLabel: "Omnibus - Add the Mattermost PPA repositories"
copyAction: "Button"

vs

event: "copy.installation"
installLabel: "Omnibus - Add the Mattermost PPA repositories"
copyAction: "Snippet"
```

**Comment by emdecr** [comment]
@cwarnermm Here's the revamped min reqs section:
http://mattermost-docs-preview-pulls.s3-website-us-east-1.amazonaws.com/6274/install/installing-mattermost-omnibus.html

For now, I've put all the raw HTML in the page template vs using a partial like we normally do for the badges.
Since we're only using this treatment for the install pages would that work? Happy to help rework in a future PR once we revamp the rest of the install pages as well.

TY~

**Comment by cwarnermm** [comment]
@emdecr - Your updates look fantastic! Thank you so much! I've confirmed with @nab-77 that over time, minimum system requirements will differ depending on the platform, so we may have up to 6 combinations in play (as expected). That's entirely manageable by the Docs team, and something we can definitely partial out in a future revision/as we go.

**Comment by emdecr** [comment]
@cwarnermm Re: button sizing - can we propose to keep the size as-is and monitor?

The thought is that the main action for the user on this page would be to focus on the install instructions (and engaging with the copy CTAs) vs contacting sales?

We've benchmarked the Contact Us conversions coming from Docs and can monitor after we push these changes.

Thoughts?

**Comment by cwarnermm** [comment]
I'm very open to publishing the CTA buttons with current sizing, and iterating based on feedback and data.

One request: Can all code examples on this Omnibus page use the new format applied to the CTA-related examples?

**Comment by emdecr** [comment]
@cwarnermm Works for us - are you talking about standalone commands on their own line vs the smaller inline ones?
Screen Shot 2023-03-21 at 1 32 42 PM

We could try a similar inline style but with just less padding. I'll start with the standalone ones first though?

File: source/install/installing-mattermost-omnibus.rst [hunk 6]
```diff
@@ -113,14 +188,52 @@ Yes. Mattermost Omnibus bundles the free, unlicensed Mattermost Enterprise Editi
...
+Can I use a custom NGINX template?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Yes. Mattermost Omnibus supports using a custom NGINX template to generate its configuration.
+
+To use this feature, you need to copy and modify the original template located at ``/opt/mattermost/mmomni/ansible/playbooks/mattermost.conf`` to a new location. Then, you can either use the variables and internal logic already bundled in the template and modify the parts that you need, or use a fully static configuration instead.
+
+After the template has been customized, add an ``nginx_template`` property to the ``/etc/mattermost/mmomni.yml`` configuration file, and then run ``mmomni reconfigure``. The reconfigure process will use the new template to generate the NGINX final configuration. You can check the contents of the ``/etc/nginx/conf.d/mattermost.conf`` file to validate that the changes were applied successfully.
+
+.. note
```
[the remark this answers is not in the document]
Comment (cwarnermm) [reply source/install/installing-mattermost-omnibus.rst]: Great catch, @emdecr! That note directive was missing required syntax (2 colons)

**Comment by cwarnermm** [comment]
The code styling looks fantastic, @emdecr! We can merge this PR when you're ready. What's left on this PR from your perspective?

**Comment by emdecr** [comment]
@cwarnermm I think we're good to merge 👍 No blockers from me. Once we merge, I'd say I'd like to take a look at the copying and tracking before we announce the switch just so I can verify that things are working and can make a patch if anything's wonky.

**Comment by cwarnermm** [comment]
@emdecr - Sounds like a plan! Would you like to do the honours and merge this one?

**Pull request merged**